### PR TITLE
feat(refactor): set_infile_info and get_fastq_file_header_info

### PR DESCRIPTION
**Pull request by henrikstranneheim**
- Added new sub add_sample_fastq_file_lanes and use it parse_sample_fastq_file_attributes
- Set file sample file info data in get_fastq_file_header_info
- Modify call to set_infile_info

### This PR adds | fixes:

- See above

### How to test:

- Automatic and continuous test suite

### Expected outcome:
- [ ] Installation, unit and integration test suite pass

### Review:
- [ ] Code review
- [ ] Tests pass

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes. If applicable record manual test results in PR header
- [ ] **MINOR** - when you add functionality in a backwards compatible manner. If applicable record manual test results in PR header
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

**Comment by jemten** [comment]
travis doesn't seem to like it. I'll hold off my review for the moment

**Comment by henrikstranneheim** [comment]
@jemten Think I fixed it

File: t/add_sample_fastq_file_lanes.t [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+add_sample_fastq_file_lanes(
+    {
+        direction      => $direction,
+        file_info_href => \%file_info,
+        lane           => $lane,
+        sample_id      => $sample_id,
+    }
+);
+
+## Then do not add lanes
```
Comment (jemten) [inline t/add_sample_fastq_file_lanes.t:93]:
```suggestion
## Then add lanes
```

File: t/set_infile_info.t [hunk 2]
```diff
@@ -71,70 +72,115 @@ my $log = test_log( {} );
 ## Set file info parameters
 
 # Interleaved files
```
Comment (jemten) [inline t/set_infile_info.t:74]:
```suggestion
```

File: lib/MIP/Fastq.pm [hunk 3]
```diff
@@ -238,16 +238,35 @@ sub get_fastq_file_header_info {
 
 ## Function : Get run info from fastq file header
 ## Returns  : @fastq_info_headers
-## Arguments: $file_path         => File path to parse
+## Arguments: $file_info_href    => File info hash {REF}
+##          : $file_name      => Fast file name
```
[the remark this answers is not in the document]
Comment (henrikstranneheim) [reply lib/MIP/Fastq.pm]: Done

File: t/add_sample_fastq_file_lanes.t (hunk 1 above)
Comment (henrikstranneheim) [reply t/add_sample_fastq_file_lanes.t]: Done

File: t/set_infile_info.t (hunk 2 above)
Comment (henrikstranneheim) [reply t/set_infile_info.t]: Done

**Pull request merged**